qiskit/qpy/common.py

```diff
@@ -20,7 +20,7 @@
 
 from qiskit.qpy import formats
 
-QPY_VERSION = 6
+QPY_VERSION = 7
```

Comment:
You don't need to bump QPY version since you didn't break any data structure.

Reply:
It feels odd not to, to me - since I defined some type keys that weren't in previous versions, we'd be breaking that property that all things written to the original version 6 spec would be able to load these new programs.  That said, the point's moot since your PR absolutely needed a version bump.

Reply:
If there is any possibility of a bug being introduced, even if it's not obvious, a version bump would help consumers track it down.